### DISCUSSION TITLE
How can i deploy my trained machine learning model on Deta?

### DISCUSSION THREAD

**Discussion by kolako54** [deta/cloud-docs#743] [Help]
I implement the API with FastAPI. and I open my trained model from the FastAPI file using Pickle.
I have no idea how to deploy the project on Deta.
this is my folder tree:

-- jobsprice.pkl
|
-- main.py
|
-- requirements.txt

and in my main.py:

```
import pickle
from pydantic import BaseModel
from fastapi import FastAPI

app = FastAPI()

model = pickle.load(open("./jobsprice.pkl", "rb"))

class Candidate(BaseModel):
    level: str
    position: str
    tech_stack: str
    experience: str
    work_type: str
    city: str

@app.post("/")
async def get_predict(data: Candidate):
        sample = [
        data.level,
        data.position,
        data.tech_stack,
        data.experience,
        data.work_type,
        data.city,
    ]
    price = model.predict(sample)
    return {
        "data": {
            "prediction": price,
        }
    }
```

**Comment by jnsougata** [comment]
First of all we need some info about the deployment size... is it over 250MB?

**Reply by kolako54** [reply]
This is less than 1mb

**Reply by jnsougata** [reply]
neat! you are good to proceed! I would recommend you to use [Deta Space](https://deta.space) . Provides 1GB of memory/invocation ..
Follow this link for more info: https://deta.space/docs/en/quickstart-guides/python
